tests/python/unit/test_frequencies.py

```diff
@@ -0,0 +1,251 @@
+        predictor=model.predictor,
+        gluon_dataset=gluon_dataset,
+        prediction_length=prediction_length,
+        quantiles=[0.5],
+        include_history=True,
+    )
+    trained_model.predict()
+    forecasts_df = trained_model.get_forecasts_df(session="2021-01-01", model_label="TEST")
+    assert forecasts_df["date"].iloc[0] == pd.Timestamp("2021-01-18")
+
+
+def test_week_sunday_frequency():
+    prediction_length = 1
+    timeseries = {
+        TIMESERIES_KEYS.START: "2021-01-17 00:00:00",
+        TIMESERIES_KEYS.TARGET: np.array([12, 13]),
+        TIMESERIES_KEYS.TARGET_NAME: "target",
+        TIMESERIES_KEYS.TIME_COLUMN_NAME: "date",
+    }
+    frequency = "W-SUN"
+    gluon_dataset = ListDataset([timeseries], freq=frequency)
+    model = Model(
+        "simplefeedforward",
+        model_parameters={"activated": True, "kwargs": {}},
+        frequency=frequency,
+        prediction_length=prediction_length,
+        epoch=1,
+        batch_size=8,
+        num_batches_per_epoch=5,
+    )
+    evaluation_forecasts_df = model.evaluate(gluon_dataset, gluon_dataset, make_forecasts=True)[2]
+    assert evaluation_forecasts_df["index"].iloc[0] == pd.Timestamp("2021-01-24")
+
+    model.train(gluon_dataset)
+    trained_model = TrainedModel(
+        predictor=model.predictor,
+        gluon_dataset=gluon_dataset,
+        prediction_length=prediction_length,
+        quantiles=[0.5],
+        include_history=True,
+    )
+    trained_model.predict()
+    forecasts_df = trained_model.get_forecasts_df(session="2021-01-01", model_label="TEST")
+    assert forecasts_df["date"].iloc[0] == pd.Timestamp("2021-01-31")
+
+
+def test_week_tuesday_frequency():
+    prediction_length = 1
+    timeseries = {
+        TIMESERIES_KEYS.START: "2021-01-19 00:00:00",
+        TIMESERIES_KEYS.TARGET: np.array([12, 13]),
+        TIMESERIES_KEYS.TARGET_NAME: "target",
+        TIMESERIES_KEYS.TIME_COLUMN_NAME: "date",
+    }
+    frequency = "W-TUE"
+    gluon_dataset = ListDataset([timeseries], freq=frequency)
+    model = Model(
+        "simplefeedforward",
+        model_parameters={"activated": True, "kwargs": {}},
+        frequency=frequency,
+        prediction_length=prediction_length,
+        epoch=1,
+        batch_size=8,
+        num_batches_per_epoch=5,
+    )
+    evaluation_forecasts_df = model.evaluate(gluon_dataset, gluon_dataset, make_forecasts=True)[2]
+    assert evaluation_forecasts_df["index"].iloc[0] == pd.Timestamp("2021-01-26")
+
+    model.train(gluon_dataset)
+    trained_model = TrainedModel(
+        predictor=model.predictor,
+        gluon_dataset=gluon_dataset,
+        prediction_length=prediction_length,
+        quantiles=[0.5],
+        include_history=True,
+    )
+    trained_model.predict()
+    forecasts_df = trained_model.get_forecasts_df(session="2021-01-01", model_label="TEST")
+    assert forecasts_df["date"].iloc[0] == pd.Timestamp("2021-02-02")
+
+
+def test_month_frequency():
```

Comment:
Can you add a comment to explain that this covers month, quarter, semester and year? 
At first, I thought we were missing those, but actually the syntax "xM" covers all.

Reply:
done in 338fa7d